src/components/edit_account/EditAccountSettings.js

```diff
@@ -0,0 +1,432 @@
+                return false;
+            }
+        }
+        return password.length === 0;
+    }
+
+    /**
+     * Return true if new password and reentered new password is equivalent
+     */
+    validatePasswordReentry() {
+        return this.state.newPasswordReentry === this.state.newPassword;
+    }
+
+    /**
+     * If emailInput is defined, return true if emailInput is valid and false otherwise.
+     * Else, check if the state of newEmail is valid, return true if valid and false otherwise.
+     * @param {String} emailInput
+     */
+    // validateEmail(email) {
+    validateEmail(emailInput) {
+        let email = emailInput || this.state.newEmail;
+        const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        return re.test(email.toLowerCase());
+    }
+
+    /**
+     * Return true if emailReentryInput is equivalent to the state of newEmail.
+     * If emailReentryInput is undefined, use the state of newEmailReentry
+     * @param {String} emailReentryInput 
+     */
+    validateEmailReentry(emailReentryInput) {
+        let emailReentry = emailReentryInput || this.state.newEmailReentry;
+        return emailReentry === this.state.newEmail;
+    }
+
+    render() {
+        let passwordWarning, newPasswordWarning, newPasswordReentryWarning, newEmailWarning, newEmailReentryWarning;
+        let passwordButtonClass = 'btn';
+        let passwordClass = 'form-group has-';
+        let newPasswordClass = 'form-group has-';
+        let newPasswordReentryClass = 'form-group has-';
+        let newEmailClass = 'form-group has-';
+        let newEmailReentryClass = 'form-group has-';
+
+        if (this.state.newPasswordIsValid !== undefined) {
+            if (this.state.newPasswordIsValid) {
+                newPasswordClass += 'success';
+            } else {
+                newPasswordClass += 'error';
+                newPasswordWarning = <span>{PASSWORD_WARNING}</span>;
+            }
+        } else {
+            newPasswordWarning = <span>{PASSWORD_WARNING}</span>;
+        }
+
+        if (this.state.newPasswordReentryIsValid !== undefined) {
+            if (this.state.newPasswordReentryIsValid) {
+                newPasswordReentryClass += 'success';
+            } else {
+                newPasswordReentryClass += 'error';
+                newPasswordReentryWarning = <span>{'Retyped password does not match new password'}</span>;
+            }
+        } else {
+            // produce initial styling when undefined
+        }
+
+        if (this.state.newEmailIsValid !== undefined) {
+            if (this.state.newEmailIsValid) {
+                newEmailClass += 'success';
+            } else {
+                newEmailClass += 'error';
+                newEmailWarning = <span>{EMAIL_WARNING}</span>;
+            }
+        } else {
+            // produce initial styling when undefined
```

Comment:
If you still need to add stuff here please do. Otherwise remove this `else` block.